drivers/overlay/overlay.go

```diff
@@ -1666,9 +1666,12 @@ func (d *Driver) get(id string, disableShifting bool, options graphdriver.MountO
 	}
 
 	mergedDir := path.Join(dir, "merged")
-	// Create the driver merged dir
-	if err := idtools.MkdirAs(mergedDir, 0o700, rootUID, rootGID); err != nil && !os.IsExist(err) {
-		return "", err
+	// attempt to create the merged dir only if it doesn't exist.
+	if _, err := os.Stat(mergedDir); err != nil {
```

Comment:
I think we need to move the `os.IsExist` check here.

Any chance we can add a test?